### DISCUSSION TITLE
Iteration output doesn't include inf_compl

### DISCUSSION THREAD

**Discussion by danieloliveira56** [coin-or/Ipopt#787] [Q&A] [answered]
Why the iteration output uses `RestoIterationOutput::WriteOutput` instead of `OrigIterationOutput::WriteOutput`?

When I run ipopt I see 
`iter    objective    inf_pr   inf_du lg(mu)...` 
instead of 
`iter  duration    objective    inf_pr   inf_du  inf_cmpl...`

Is it possible to use `OrigIterationOutput`?

**Comment by svigerske** [comment, marked as answer]
I couldn't find any mention of `duration` in the Ipopt source.
The headers for `OrigIterationOutput` and `RestoIterationOutput` are the same:
https://github.com/coin-or/Ipopt/blob/e10e5c738605f0525a50cbdaa624d56378986c77/src/Algorithm/IpOrigIterationOutput.cpp#L81
https://github.com/coin-or/Ipopt/blob/e10e5c738605f0525a50cbdaa624d56378986c77/src/Algorithm/IpRestoIterationOutput.cpp#L71

You should get the output from `RestoIterationOutput::WriteOutput` during restoration phase, and the output from `OrigIterationOutput::WriteOutput` outside restoration phase.

The `RestoIterationOutput::WriteOutput` seems to allow to call also `OrigIterationOutput::WriteOutput`. There is no option for that, but you would need to change this code to create `resto_OrigIterOutput`:

https://github.com/coin-or/Ipopt/blob/e10e5c738605f0525a50cbdaa624d56378986c77/src/Algorithm/IpAlgBuilder.cpp#L980-L983

To get always only OrigIterationOutput, you could try changing this code to
```cpp
SmartPtr<IterationOutput> resto_IterOutput = new OrigIterationOutput();
```

**Reply by danieloliveira56** [reply]
I'm so sorry, I edited the header some time ago for testing and completely forgot.
